BrowserKit/Sources/MenuKit/MenuTableView.swift

```diff
@@ -5,12 +5,19 @@
 import Foundation
 import UIKit
 
-class MenuTableView: UIView, UITableViewDataSource, UITableViewDelegate {
-    private let tableView: UITableView
-    private var menuData: [MenuSection] = []
+public protocol MenuTableViewDataDelegate: AnyObject {
+    func reloadTableView(with data: [MenuSection])
```

Comment:
instead of reloading via the delegate, could we update the main menu state and by updating the state, then we call the reload?

Reply:
yep. That'll be the next PR to add Redux to the DetailVC